### PR TITLE
Move effects

### PR DESCRIPTION
**Pull request by st-pasha**
# Description

Added new `MoveEffect` class based on `Transform2DEffect`. Compared to the original `MoveEffect`, these are some differences:
  - `MoveEffect.to` implements movement towards a specific point in the world;
  - `MoveEffect.by` is a relative motion that changes the position of a component by the given offset;
  - `MoveEffect.along` is for moving the component along an arbitrary path (which includes curved paths too);
  - multiple move effects can be applied to a component at the same time.

Added new example in the Dashbook demonstrating different kinds of move effects.

https://user-images.githubusercontent.com/744771/140383592-36a9c090-f6f8-40a9-804e-cb405d29d89d.mp4

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `./scripts/format.sh` and the Flame analyzer (`./scripts/analyze.sh`) does not report any problems.
- [x] I read and followed the [Flame Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

- [x] No, this is *not* a breaking change.

## Related Issues

Replaces #1031

### CODE REVIEW

File: examples/lib/stories/effects/move_effect_example.dart [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import 'dart:math';
+
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+import 'package:flame/src/effects2/move_effect.dart'; // ignore: implementation_imports
+import 'package:flame/src/effects2/standard_effect_controller.dart'; // ignore: implementation_imports
```
Comment (spydon) [inline examples/lib/stories/effects/move_effect_example.dart:6]: We'll have to remember to remove these ignores later when we export everything from the new effects system.

File: examples/lib/stories/effects/move_effect_example.dart [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      add(
+        SquareComponent(size: 10)
+          ..paint = (Paint()..color = const Color(0xFF00FFF7))
+          ..add(MoveEffect.along(
+              path2,
+              StandardEffectController(
+                duration: 6,
+                startDelay: i * 0.3,
+                infinite: true,
+              ))),
```
Comment (spydon) [inline examples/lib/stories/effects/move_effect_example.dart:104]: Some more trailing commas here might make it clearer which parenthesis is for which start parenthesis

File: packages/flame/lib/src/effects2/move_effect.dart [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  late final PathMetric _pathMetric;
+  late final double _pathLength;
+  late Vector2 _lastPosition;
+
+  @override
+  void apply(double progress) {
+    final distance = progress * _pathLength;
+    final tangent = _pathMetric.getTangentForOffset(distance)!;
+    final offset = tangent.position;
+    final newPosition = Vector2(offset.dx, offset.dy);
```
Comment (spydon) [inline packages/flame/lib/src/effects2/move_effect.dart:87]: If you move the creation of this `Vector2` outside of `apply` we don't have to create new objects in every tick for it, and we can just use `.setValues` instead.

File: packages/flame/lib/src/effects2/move_effect.dart [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+class MoveEffect extends Transform2DEffect {
+  MoveEffect.by(Vector2 offset, EffectController controller)
+      : _offset = offset.clone(),
+        super(controller);
+
+  factory MoveEffect.to(Vector2 destination, EffectController controller) =>
+      _MoveToEffect(destination, controller);
+
+  factory MoveEffect.along(Path path, EffectController controller) =>
+      _MoveAlongPathEffect(path, controller);
```
Comment (spydon) [inline packages/flame/lib/src/effects2/move_effect.dart:36]: Awesome with support for `Path`!

**Review by spydon — APPROVED** [review]
Nice, really looking forward to be able to pass `Path`s to the `MoveEffect`.

File: packages/flame/lib/src/effects2/move_effect.dart (hunk 3 above)
Comment (st-pasha) [reply packages/flame/lib/src/effects2/move_effect.dart:87]: ok

File: packages/flame/lib/src/effects2/move_effect.dart (hunk 4 above)
Comment (st-pasha) [reply packages/flame/lib/src/effects2/move_effect.dart:36]: check out the example too

File: examples/lib/stories/effects/move_effect_example.dart (hunk 2 above)
Comment (st-pasha) [reply examples/lib/stories/effects/move_effect_example.dart]: added

**Review by erickzanardo — APPROVED** [review]

**Pull request merged**